tasks/programs_and_groups.yml

```diff
@@ -21,17 +21,19 @@
   when: supervisor_program_groups|length > 0
   notify: restart supervisor managed programs
 
-# TODO: use find module once we move to 2.0
 - name: Find conf files in supervisor_include_path
-  shell: find {{supervisor_include_path}} -name '*.conf' -type f -printf '%f\n'|sed 's/.conf//'
+  find:
+    paths: "{{supervisor_include_path}}"
+    patterns: '*.conf'
+    file_type: 'file'
   register: all_supervisor_confs
 
 - name: Remove un-managed supervisor config files
   file:
-    path: "{{supervisor_include_path}}/{{item}}.conf"
+    path: "{{ item.path }}"
     state: absent
-  with_items: "{{all_supervisor_confs.stdout_lines | default([])}}"
+  with_items: "{{all_supervisor_confs.files | default({})}}"
```

Comment:
`with_items` is for looping over lists. Depending on what `all_supervisor_confs.files` returns, this should either default to `[]` or use a `with_dict:`.